tests/test_script_queue.py

```diff
@@ -348,6 +349,126 @@ def make_add_data(location, locationSalIndex=0):
+            # else check that no second log level was output
+            if log_level != 0:
+                data = await script_remote.evt_logLevel.next(flush=False, timeout=STD_TIMEOUT)
+                self.assertEqual(data.level, log_level)
+            else:
+                with self.assertRaises(asyncio.TimeoutError):
+                    await script_remote.evt_logLevel.next(flush=False, timeout=0.01)
+
+            # make sure the script queue knows that the script is ready to run
+            await self.wait_runnable(I0)
+
+            # enable the queue and let it run
+            await self.remote.cmd_resume.start(timeout=STD_TIMEOUT)
+            await self.assert_next_queue(enabled=True, running=True, currentSalIndex=I0)
+            await self.assert_next_queue(enabled=True, running=True, pastSalIndices=[I0])
+
+    async def test_add_nonzero_log_level(self):
+        """Test addding a script with a non-zero log level."""
+        # pick a level that does not match the default
+        # to make it easier to see that the level has changed
+        log_level = logging.INFO - 1
+        await self.check_add_log_level(log_level=log_level)
+
+    async def test_add_zero_log_level(self):
+        """Test addding a script with log level 0, meaning don't change it."""
+        await self.check_add_log_level(log_level=0)
+
+    async def test_add_and_pause(self):
+        """Test adding a script with a pause checkpoint.
+        """
+        await self.assert_next_queue(enabled=False, running=True)
+        await self.remote.cmd_enable.start(timeout=STD_TIMEOUT)
+        await self.assert_next_queue(enabled=True, running=True)
+
+        async with salobj.Remote(domain=self.queue.domain, name="Script", index=I0) as script_remote:
+            await self.remote.cmd_add.set_start(
+                pauseCheckpoint="start",
+                isStandard=False,
+                path="script1",
+                config="",
+                location=Location.LAST,
+                descr="test_add",
+                timeout=START_TIMEOUT)
+            await self.assert_next_queue(running=True, salIndices=[I0])
+            await self.assert_next_queue(running=True, currentSalIndex=I0)
+            timeout = START_TIMEOUT
+            for expected_script_state in (ScriptState.UNCONFIGURED,
+                                          ScriptState.CONFIGURED,
+                                          ScriptState.RUNNING,
+                                          ScriptState.PAUSED):
+                script_state = await script_remote.evt_state.next(flush=False, timeout=timeout)
+                self.assertEqual(script_state.state, expected_script_state)
+                timeout = STD_TIMEOUT
+
+            await script_remote.cmd_resume.start(timeout=STD_TIMEOUT)
+            await self.assert_next_queue(running=True, pastSalIndices=[I0])
+
+    async def test_add_and_stop(self):
+        """Test adding a script with a stop checkpoint.
+        """
```

Comment:
I have the same comments here as I did for the pause tests above re: how this works.

Reply:
Agreed. I hope the new comments will suffice.